Add description on building backend for development

BranAndSceolan opened the pull request:
Add badge showing codeql-analysis result

BranAndSceolan committed:
Add description on building backend for development
Add badge showing codeql-analysis result

danielp0505 commented on README.md:
```diff
@@ -1,3 +1,27 @@
+  * build for development: 
+    * create a .env file with
+      * NOTES_PASSWORD=choose_a_password
+      * POSTGRES_USER=choose_an_admin_name
```
Why does the backend need/use the admin account?

BranAndSceolan replied:
The init-script used in the Dockerfile uses the admin account, so I thought backend needs it to construct the database container.

danielp0505 commented on README.md:
```diff
@@ -1,3 +1,27 @@
+      * NODE_ENV=debug
+      * NOTES_USER=choose_a_username
+    * Replace all choose_-entries with the values you would like to use.
+    * source your .env file  ``source <path to .env>``
```
This is an incomplete command, is it a parameter?

BranAndSceolan replied:
yes, <path to .env> is intended as a placeholder for a parameter here.

danielp0505 commented on README.md:
```diff
@@ -1,3 +1,27 @@
+      * NOTES_USER=choose_a_username
+    * Replace all choose_-entries with the values you would like to use.
+    * source your .env file  ``source <path to .env>``
+    * execute the docker-compose file in the root directory. ``docker-compose build``
```
This just builds images and doesn't start a database container if this is what you're referring to here. https://docs.docker.com/engine/reference/commandline/compose_build/
docker-compose up should automatically build the images and create containers as specified in the docker-compose.yml.

BranAndSceolan replied:
I replaced that part with cocker-compose up like recommended

danielp0505 reviewed (changes requested):
The instructions are incomplete/incorrect. Please see the comments.

BranAndSceolan committed:
Merge branch 'development' into 16-update-readmemd

BranAndSceolan committed:
change docker-compose build to docker compose up

BranAndSceolan committed:
updated instructions

updated instructions on creating a development build and replacing an old development build with a new one

BranAndSceolan committed:
remove unused config files and explain how to use remote chai test

BranAndSceolan committed:
add instructions on building the backend for deployment

BranAndSceolan committed:
Merge branch 'development' into 16-update-readmemd

# Conflicts:
#	backend/config/deployment.json
#	backend/modules/controllers/documents.controller.ts

BranAndSceolan committed:
remove string tranformation from create in documents.controller.ts

BranAndSceolan committed:
add instructions on how to start local deployment build

BranAndSceolan committed:
Merge branch 'development' into 16-update-readmemd

# Conflicts:
#	README.md
